Makefile

```diff
@@ -0,0 +1,71 @@
+# A simple Makefile intended to facilitate repetitive tasks
+
+# Dependencies installation
+.PHONY: dep-install
+dep-install:
+	@echo "#### Installing dependencies ####"
+	pip install numpy
+	pip install scipy
+	pip install scikit-learn
+	# Used to run the examples
+	pip install matplotlib
+	# Package needed to run tests in an automate way
+	pip install nose
+	@echo "#### All dependencies were installed successfully ####"
+
+# Test related commands
+NOSETESTS ?= nosetests
+
+.PHONY: test-all
+test-all:
+	nosetests
+
+# Examples related commands
+.PHONY: run-poisson
+run-poisson:
+	python examples/api/plot_poisson.py
+
+.PHONY: run-poisson2
```

Comment:
On Wed, May 25, 2016 at 6:00 PM, Vinicius Marques <notifications@github.com>
wrote:

> In Makefile
> <https://github.com/pavanramkumar/pyglmnet/pull/115#discussion_r64598704>:
>
> > +	pip install nose
> > +	@echo "#### All dependencies were installed successfully ####"
> > +
> > +# Test related commands
> > +NOSETESTS ?= nosetests
> > +
> > +.PHONY: test-all
> > +test-all:
> > +	nosetests
> > +
> > +# Examples related commands
> > +.PHONY: run-poisson
> > +run-poisson:
> > +	python examples/api/plot_poisson.py
> > +
> > +.PHONY: run-poisson2
>
> Matplotlib package is needed to build the documentation. Thus, I suggest
> to add a command line to install it on the doc-dependencies target.
>

sounds good to me.

> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/pavanramkumar/pyglmnet/pull/115/files/a01cd7ee566abf72de9d4af95f02df2171a336c0#r64598704>
>